contracts/dev-utils/contracts/src/OrderValidationUtils.sol

```diff
@@ -185,4 +197,47 @@ contract OrderValidationUtils is
         transferableAssetAmount = LibSafeMath.min256(balance, allowance);
         return transferableAssetAmount;
     }
+
+    /// @dev This function handles the edge cases around taker validation. This function
+    ///      currently attempts to find duplicate ERC721 token's in the taker
+    ///      multiAssetData.
+    /// @param assetData The asset data that should be validated.
+    /// @return Whether or not the order should be considered valid.
+    function _isAssetDataValid(bytes memory assetData)
+        internal
+        pure
+        returns (bool)
+    {
+        // Only process the taker asset data if it is multiAssetData.
+        bytes4 assetProxyId = assetData.readBytes4(0);
```

Comment:
Nit: we should probably check that `assetData.length > 3 && assetData%4 == 0` before doing anything else here.

Reply:
Shouldn't this condition be `assetData.length > 3 && assetData % 32 == 4`?